<span style="color:red;font-style:normal;font-family:Georgia">1. IMPORTAMOS LAS LIBRERIAS.</span>

In [17]:
import pandas as pd
import numpy as np
import matplotlib.pylab as plt
import seaborn as sns
from scipy.stats import shapiro
from statsmodels.stats.diagnostic import lilliefors
from scipy import stats

<span style="color:red;font-style:normal;font-family:Georgia">2. IMPORTAMOS LA DATA DE NUESTRA MUESTRA.</span>

In [2]:
data=pd.read_excel("caso1.xlsx")
data.head()

Nombre  Antes  Después
0  Estudiante 1     18       22
1  Estudiante 2     21       25
2  Estudiante 3     16       17
3  Estudiante 4     22       24
4  Estudiante 5     19       16

<span style="color:red;font-style:normal;font-family:Georgia">3. PARA UTILIZAR LA PRUEBA T PARA MUESTRAS PAREADAS PRIMERO DEBEN CUMPLIRSE CIERTOS REQUISITOS.</span>

![](https://lh3.googleusercontent.com/proxy/3Bpbm0AFxwo9I7mlg2NAc13m8896EJCXGxv-xL_YV0PJ8SGSPWkDMjF8QktAlr6rY_tGjIvrQtMiQuUcyssy4g4R3M3nhqlTA4Lf66radzMU9jYsxFW1_YBjN_8aflDYF7f0svAZoAfOvpeG5MEZChGO-VV4hEyFQA)

<span style="color:red;font-style:normal;font-family:Georgia">3.1 LAS VARIABLES DEBEN SER CUANTITATIVAS(DISCRETAS O CONTINUAS).</span>

In [22]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20 entries, 0 to 19
Data columns (total 3 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   Nombre   20 non-null     object
 1   Antes    20 non-null     int64 
 2   Después  20 non-null     int64 
dtypes: int64(2), object(1)
memory usage: 608.0+ bytes


<span style="color:red;font-style:normal;font-family:Georgia">CONCLUCIÓN :</span>

- Vemos que ambas variables son cuantitativas discretas, de este modo se cumpliría la condición.

<span style="color:red;font-style:normal;font-family:Georgia">3.2 LAS VARIABLES EN ESTUDIO DEBEN SEGUIR UN COMPORTAMIENTO NORMAL. PARA COMPROBAR ESTA CONDICION FORMULAREMOS UNA HIPOTESIS CON SUS RESPECTIVAS PRUEBAS Y UN NIVEL DE SIGNIFICANCIA DEL 5%.</span>

<span style="color:black;font-style:normal;font-family:Georgia">HIPOTESIS :</span>

- <span style="color:black;font-style:normal;font-family:Georgia">Ho : La variable en estudio tiene una Distribución Normal.</span>
- <span style="color:black;font-style:normal;font-family:Georgia">H1 : La variable en estudio no tienen una Distribución Normal.</span>

## **Prueba de Shapiro-Wilk**

In [5]:
prueba_nor_antes=shapiro(data["Antes"])
prueba_nor_antes

ShapiroResult(statistic=0.9819655418395996, pvalue=0.9568763375282288)

- <span style="color:black;font-style:normal;font-family:Georgia">Como 0.9569 > 0.05 la variable Antes seguiría un comportamiento normal.</span>

In [6]:
prueba_nor_despues=shapiro(data["Después"])
prueba_nor_despues

ShapiroResult(statistic=0.9423450231552124, pvalue=0.2654399871826172)

- <span style="color:black;font-style:normal;font-family:Georgia">Como 0.2654 > 0.05 la variable Después seguiría un comportamiento normal.</span>

## **Prueba de lilliefors**

In [8]:
prueba_lilliefors_antes=lilliefors(data["Antes"])
prueba_lilliefors_antes

C:\Users\pc\anaconda3\lib\site-packages\statsmodels\stats\_lilliefors.py:272: FutureWarning: The default pvalmethod will change from "approx" to "table" after 0.11. The "table" method uses values from a very large simulation and is more accurate.  Explicitly set this parameter to "approx" or "table" to silence this warning
  warnings.warn(msg, FutureWarning)


(0.10048724045280688, 0.8509813199711381)

- <span style="color:black;font-style:normal;font-family:Georgia">Como 0.85098 > 0.05 la variable Antes seguiría un comportamiento normal.</span>

In [9]:
prueba_lilliefors_despues=lilliefors(data["Después"])
prueba_lilliefors_despues

(0.14414410746424744, 0.3391498239042544)

- <span style="color:black;font-style:normal;font-family:Georgia">Como 0.3391 > 0.05 la variable Después seguiría un comportamiento normal.</span>

<span style="color:black;font-style:normal;font-family:Georgia">CONCLUCIÓN :</span>

- <span style="color:black;font-style:normal;font-family:Georgia">Hay evidencia estadística suficiente para decir que la variable en estudio proviene de una distribución normal.</span>

<span style="color:red;font-style:normal;font-family:Georgia">4. COMO SE CUMPLEN LAS CONDICIONES PODEMOS APLICAR LA PRUEBA T PARA MUESTRAS PAREADAS. PARA ESTO FORMULAMOS LA HIPOTESIS CON UN NIVEL DE SIGNIFICANCIA DEL 5%.</span>

<span style="color:black;font-style:normal;font-family:Georgia">HIPOTESIS :</span>

- <span style="color:black;font-style:normal;font-family:Georgia">H0 : Los puntajes son iguales.</span>
- <span style="color:black;font-style:normal;font-family:Georgia">H1 : El puntaje antes es menor al puntaje después.</span>

In [18]:
stats.ttest_rel(data["Antes"],data["Después"])

Ttest_relResult(statistic=-3.231252665580312, pvalue=0.004394965993185664)

<span style="color:black;font-style:normal;font-family:Georgia">CONCLUCIÓN :</span>

- <span style="color:black;font-style:normal;font-family:Georgia">Como es una prueba bilateral primero debemos dividir el pvalue/2 entonces 0.0022 < 0.05 rechazamos H0.</span>
- <span style="color:black;font-style:normal;font-family:Georgia">Hay evidencia estadística para afirmar que el puntaje antes es menor al puntaje después.</span>